components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/MonitoredItemMessageEncoder.cs

```diff
@@ -316,30 +322,43 @@ public class MonitoredItemMessageEncoder : IMessageEncoder {
+                        try {
+                            result = new MonitoredItemMessage {
+                                MessageContentMask = (message.Writer?.MessageSettings?
                                .DataSetMessageContentMask).ToMonitoredItemMessageMask(
                                    message.Writer?.DataSetFieldContentMask),
-                            ApplicationUri = message.ApplicationUri,
-                            EndpointUrl = message.EndpointUrl,
-                            ExtensionFields = message.Writer?.DataSet?.ExtensionFields,
-                            NodeId = notification.NodeId.ToExpandedNodeId(message.ServiceMessageContext?.NamespaceUris),
-                            Timestamp = message.TimeStamp ?? DateTime.UtcNow,
-                            Value = notification.Value,
-                            DisplayName = notification.DisplayName,
-                            SequenceNumber = notification.SequenceNumber.GetValueOrDefault(0)
-                        };
-                        // force published timestamp into to source timestamp for the legacy heartbeat compatibility
-                        if (notification.IsHeartbeat &&
-                            ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.Timestamp) == 0) &&
-                            ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.SourceTimestamp) != 0)) {
-                            result.Value.SourceTimestamp = result.Timestamp;
+                                ApplicationUri = message.ApplicationUri,
+                                EndpointUrl = message.EndpointUrl,
+                                ExtensionFields = message.Writer?.DataSet?.ExtensionFields,
+                                NodeId = notification.NodeId.ToExpandedNodeId(message.ServiceMessageContext?.NamespaceUris),
+                                Timestamp = message.TimeStamp ?? DateTime.UtcNow,
+                                Value = notification.Value,
+                                DisplayName = notification.DisplayName,
+                                SequenceNumber = notification.SequenceNumber.GetValueOrDefault(0)
+                            };
+                            // force published timestamp into to source timestamp for the legacy heartbeat compatibility
+                            if (notification.IsHeartbeat &&
+                                ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.Timestamp) == 0) &&
+                                ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.SourceTimestamp) != 0)) {
+                                result.Value.SourceTimestamp = result.Timestamp;
+                            }
+
+                        }
+                        catch (Exception e) {
+                            _logger.Error(e, $"Error in notification {notificationCounter}/{message.Notifications.Count()}");
```

Comment:
Structured logging please.   Also, do we want to encode the error for diagnostics into the message or so?

Reply:
Done incl. e.Message